roles/DCOS.requirements/tasks/main.yml

```diff
@@ -23,6 +23,14 @@
     enabled: false
   when: "'dnsmasq.service' in ansible_facts.services"
 
+- name: Check if docker is already installed
+  yum:
+    list:
+      - docker
+      - docker-ce
+  # If not installed yum_list.results[*].yumstate != installed
+  register: yum_list
```

Comment:
i think `yum_list` is to unspecific. In this particular case at least something like `yum_list_docker` should be used.